### PR TITLE
Add AzureCLI 2.42.0

### PR DESCRIPTION
**Pull request by azclibot**
Add AzureCLI 2.42.0

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/87039)

### CODE REVIEW

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+PackageIdentifier: Microsoft.AzureCLI
+PackageVersion: 2.42.0
+PackageName: Microsoft Azure CLI
+Publisher: Microsoft Corporation
+Moniker: azure-cli
+PackageUrl: https://docs.microsoft.com/en-us/cli/azure/
+License: Copyright (C) Microsoft Corporation
```
Comment (Trenly) [inline manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:7]: A copyright isn't a license; Please update this with actual license terms such as Proprietary, MIT, GPL-2.0, or other terms as appropriate 
```suggestion
License: Proprietary
Copyright: Copyright (C) Microsoft Corporation
```

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ShortDescription: The Azure command-line interface (Azure CLI) is a set of commands used to create and manage Azure resources. The Azure CLI is available across Azure services and is designed to get you working quickly with Azure, with an emphasis on automation.
+Installers:
+- Architecture: x64
+  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.42.0.msi
+  InstallerType: msi
+  InstallerSha256: b8ce13727e044efdf350e4d0bd49ce2aee418fcc6a1c10fe52ad046a5c97b521
+PackageLocale: en-US
+ManifestType: singleton
```
Comment (Trenly) [inline manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:15]: Please move to the Multi-manifest format

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ManifestVersion: 1.0.0
```
Comment (Trenly) [inline manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:16]: Please move to Manifest version 1.2.0

**Review by Trenly — CHANGES_REQUESTED** [review]

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml (hunk 3 above)
Comment (wangzelin007) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:16]: @Trenly,
Please help close this PR, we will re-issue a PR to cover this PR.
Is there any relevant doc to guide how to upgrade to Manifest version 1.2.0?
Can we upgrade to Manifest version 1.2.0 in the next version?

**Comment by wangzelin007** [comment]
@Trenly，
I saw that the Installation Validation failed, but there was no error log. Can you please take a look at it?
https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=393159&view=logs&j=871b4eed-22de-5cb6-c1c8-82f190de633f&t=58bdf892-4e45-5ba7-4782-31912f95e167
image

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml (hunk 3 above)
Comment (wangzelin007) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:16]: @Trenly,
Actually, this file is hardcoded, can we simply modify ManifestVersion from 1.0.0 to 1.2.0?

Comment (Trenly) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:16]: > @Trenly,
> Actually, this file is hardcoded, can we simply modify ManifestVersion from 1.0.0 to 1.2.0?

Yes, that should work

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+License: Copyright (C) Microsoft Corporation
+ShortDescription: The Azure command-line interface (Azure CLI) is a set of commands used to create and manage Azure resources. The Azure CLI is available across Azure services and is designed to get you working quickly with Azure, with an emphasis on automation.
+Installers:
+- Architecture: x64
+  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.42.0.msi
+  InstallerType: msi
+  InstallerSha256: 70e18e9f782c29e482a10661f97a1449dea7d681ac559aad8f3f33e32961e74c
+PackageLocale: en-US
+ManifestType: singleton
+ManifestVersion: 1.0.0
```
Comment (wangzelin007) [inline manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:16]:
```suggestion
ManifestVersion: 1.2.0
```

File: manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml (hunk 2 above)
Comment (wangzelin007) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:15]: Hi @Trenly 
Is there any relevant doc or PR to guide how to use Multi-manifest format?

Comment (Trenly) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:15]: https://learn.microsoft.com/en-us/windows/package-manager/package/manifest?tabs=minschema%2Cversion-example#multiple-manifest-files

Using [wingetcreate](https://github.com/microsoft/winget-create) for your bot integrations or [CI/CD](https://github.com/microsoft/winget-create) will automatically use multi-manifest and latest ManifestVersion

Comment (wangzelin007) [reply manifests/m/Microsoft/AzureCLI/2.42.0/Microsoft.AzureCLI.yaml:15]: Hi @Trenly
I will add wingetcreate for our bot integrations later.
Please close this PR and move to #87136 which create by wingetcreate.

**Comment by stephengillie** [comment]
> Please close this PR and move to #87136 which create by wingetcreate.
Duplicate of #87136

**Pull request closed without merging**